dashboard/app/controllers/followers_controller.rb

```diff
@@ -49,6 +67,41 @@ def remove
     )
   end
 
+  def remove_async
+    @section = Section.find_by_code params[:section_code]
+
+    if @section
+      f = Follower.where(section: @section.id, student_user_id: current_user.id).first
+    end
+
+    unless @section && f
+      render json: {
+        result: "error",
+        sections: current_user.sections_as_student.map(&:summarize)
+      }
+      return
+    end
+
+    @teacher = @section.user
+
+    authorize! :destroy, f
+    f.delete
```

Comment:
+1!

Perhaps the notify should be in the model behind a flag, something like
```
section.remove_student(student, notify: true)
```

Also consider using a [serializer](https://github.com/code-dot-org/code-dot-org/tree/44d3d82e55ac8d97fa0a3fa2be42e62c214ecddc/dashboard/app/serializers/api/v1) for the json.